pkg/github/github.go

```diff
@@ -0,0 +1,56 @@
+package github
+
+import (
+	"context"
+
+	gogit "github.com/google/go-github/v60/github"
+
+	"k8s.io/utils/ptr"
+
+	"sigs.k8s.io/release-sdk/github"
+)
+
+func UploadSBOMAsset(ctx context.Context, owner, repo, tag, sbomFile string) error {
+	client := github.New()
+
+	release, res, err := client.Client().GetReleaseByTag(ctx, owner, repo, tag)
+	if err != nil && res.StatusCode != 404 {
+		return err
+	}
+
+	// maybe the release is in draft mode
+	if release == nil {
+		releases, err := client.Releases(owner, repo, true)
+		if err != nil && res.StatusCode != 404 {
+			return err
+		}
+
+		for _, r := range releases {
+			if tag == r.GetTagName() {
+				release = r
+				break
+			}
+		}
+	}
+
+	// if we still can't find the release, create a new one
+	if release == nil {
+		opts := &gogit.RepositoryRelease{
```

Comment:
I'm not sure we should be creating the release here.

It does feel a little intrusive and it would be surprising and annoying to see that the SBOM tool created a release if you weren't expecting it.

What do you think if we just upload the SBOM to an existing one and fail if it's not there?

My concern is that we would need to add many other features to make this work properly (draft/nodraft, release name, prelelase, etc). And we already have that in publish-release.